lib/inferno/dsl/fhir_evaluation/rules/all_references_resolve.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require_relative '../../../utils/evaluator_util'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      module Rules
+        class AllReferencesResolve < Rule
+          def check(context)
+            # resource_type_ids is for quick look up when there is a reference type
+            # resource_ids is for quick look up when there is no type (i.e. uuid used)
+            util = Inferno::Utils::EvaluatorUtil
```

Comment:
I know I've looked at this before and worked my way through it, but it's not easy to follow quickly. Is it possible to clean this up at all?

Reply:
Cleaned up the rules and extractor codes overall.